README.md

```diff
@@ -320,6 +320,77 @@ Input variable structure as in good example 2 is preferred over example 1 where
 
   Reference - [https://learn.hashicorp.com/tutorials/terraform/outputs](https://learn.hashicorp.com/tutorials/terraform/outputs)
 
+### 2.6 Do not use module calls as objects
+
+Do not refer to the entire module call as a one object, such as `x = module.mystuff`. Although it works on `terraform apply`, it propagates deletions unexpectedly on all Terraform versions. Example code which uses `random_pet`, a very simple Hashicorp-provided module which simply generates names, e.g. "relevant-horse":
+
+* producer/main.tf
+
+      ```hcl2
+      resource "random_pet" "unrelated" {
+        # It uses nothing, and is not used by anything. Right?
+      }
+
+      resource "random_pet" "this" {}
+
+      output "id" {
+        value = random_pet.this.id
+      }
+      ```
+
+* main.tf
+
+      ```hcl2
+      module "producer" {
+        source = "./producer"
+      }
+
+      locals {
+        # BAD. Whether you pass the module.producer through a `local`, a `variable`, an `output`, the behavior is not great.
+        module_call_as_object = module.producer
+      }
+
+      resource "random_pet" "main" {
+        prefix = local.module_call_as_object.id
+      }
+      ```
+
+So, what is the problem? When any Terraform run tries to delete the `unrelated` resource, it does delete also `main` resource, despite
+there is no apparent dependency between them:
+
+    ```sh
+    $ terraform apply
+    $ terraform destroy -target module.producer.random_pet.unrelated
+
+    Terraform will perform the following actions:
+
+      # random_pet.main will be destroyed
+      - resource "random_pet" "main" {
+          - id        = "dynamic-basilisk-evident-bat" -> null
+          - length    = 2 -> null
+          - prefix    = "dynamic-basilisk" -> null
+          - separator = "-" -> null
+        }
```

Comment:
The crux of the problem. This entire section is about how to avoid this deletion.